controllers/arm.py

```diff
@@ -0,0 +1,35 @@
+from components.arm import Arm, Setpoint, Setpoints
+
+from magicbot import StateMachine, state
+
+
+class ArmController(StateMachine):
+    arm: Arm
+
+    def __init__(self) -> None:
+        self._target_setpoint: Setpoint = Setpoints.STOW
+
+    def go_to_setpoint(self, setpoint: Setpoint) -> None:
+        # Only restart the state machine if the setpoint is different
+        if setpoint != self._target_setpoint:
+            self.done()
+            self._target_setpoint = setpoint
+            self.engage()
+
+    def get_angle(self) -> float:
+        return self.arm_component.get_angle()
+
+    def at_goal(self) -> bool:
+        return not self.is_executing
+
+    @state(first=True, must_finish=True)
+    def retract_arm(self) -> None:
+        self.next_state("rotate_arm")
```

Comment:
```suggestion
    def retracting_arm(self) -> None:
        self.next_state("rotating_arm")
```
Dont we generally try to name all of our states with "ing" so we know they are doing something and its not part of the interface to tell it to do it

Reply:
Yep!